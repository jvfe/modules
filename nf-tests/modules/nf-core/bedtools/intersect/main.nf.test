nextflow_process {

    name "Test Process BEDTOOLS_INTERSECT"
    script "modules/nf-core/bedtools/intersect/main.nf"
    process "BEDTOOLS_INTERSECT"
    config "nf-tests/modules/nf-core/bedtools/intersect/nextflow.config"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:

            }
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['test2_bed'], checkIfExists: true)
                           ]
                input[1] = 'bed'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should run with bam file") {

        when {
            params {
                // define parameters here. Example:

            }
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['genome']['baits_bed'], checkIfExists: true)
                           ]
                input[1] = 'bam'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
