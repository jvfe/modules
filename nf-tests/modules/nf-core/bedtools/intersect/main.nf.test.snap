{
    "Should run without failures": {
        "content": [
            {
                "intersect": [
                    [
                        {
                            "id": "test"
                        },
                        "test_out.bed:md5,afcbf01c2f2013aad71dbe8e34f2c15c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b75347c6df1358fdf017dbe47c18af7f"
                ]
            }
        ],
        "timestamp": "2022-11-15T15:24:30+0000"
    },
    "Should run with bam file": {
        "content": [
            {
                "intersect": [
                    [
                        {
                            "id": "test"
                        },
                        "test_out.bam:md5,738324efe2b1e442ceb6539a630c3fe6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b75347c6df1358fdf017dbe47c18af7f"
                ]
            }
        ],
        "timestamp": "2022-11-15T15:24:30+0000"
    }
}